core/src/main/java/io/fabric8/maven/core/config/ProcessorConfig.java

```diff
@@ -121,24 +83,117 @@ public boolean use(String name) {
         }
         for (String inc : includes) {
-            T named = lookup.get(inc);
-            if (named == null) {
-                throw new IllegalArgumentException("No " + type + " with name '" + inc +
-                                                   "' found to include. " +
-                                                   "Please check spelling and your project dependencies");
+            if (use(inc)) {
+                T named = lookup.get(inc);
+                if (named == null) {
+                    List<String> keys = new ArrayList<>(lookup.keySet());
+                    Collections.sort(keys);
+                    throw new IllegalArgumentException(
+                        "No " + type + " with name '" + inc +
+                        "' found to include. " +
+                        "Please check spelling in your profile / config and your project dependencies. Included " + type + "s: " +
+                        StringUtils.join(keys,", "));
+                }
+                ret.add(named);
+            }
+        }
+        return ret;
+    }
+
+    public boolean use(String inc) {
+        return !excludes.contains(inc) && includes.contains(inc);
+    }
+
+    /**
+     * Merge in another processor configuration, with an higher priority. This operation mutates
+     * the current config.
+     *
+     * @param config config to merge in
+     * @return this merged configuration for convenience of chaining and returning.
+     */
+    public static ProcessorConfig mergeProcessorConfigs(ProcessorConfig ... processorConfigs) {
+        // Merge the configuration
+        Map<String, TreeMap> configs = mergeConfig(processorConfigs);
+
+        // Get all includes
+        Set<String> excludes = mergeExcludes(processorConfigs);
+
+        // Find the set of includes, which are the ones from the profile + the ones configured
+        List<String> includes = mergeIncludes(processorConfigs);
+
+        return new ProcessorConfig(includes, excludes, configs);
+    }
+
+    private static Set<String> mergeExcludes(ProcessorConfig ... configs) {
+        Set<String> ret = new HashSet<>();
+        for (ProcessorConfig config : configs) {
+            if (config != null) {
+                Set<String> excludes = config.excludes;
+                if (excludes != null) {
+                    ret.addAll(excludes);
+                }
             }
-            ret.add(named);
         }
         return ret;
     }
+
+    private static List<String> mergeIncludes(ProcessorConfig ... configs) {
+        List<String> ret = new ArrayList<>();
+        for (ProcessorConfig config : configs) {
+            if (config != null) {
+                List<String> includes = config.includes;
+                if (includes != null) {
+                    ret.addAll(includes);
+                }
+            }
+        }
+        return removeDups(ret);
+    }
+
+    // Remove duplicates such that the later element remains
```

Comment:
AFAIK does LinkedHashSet not change the insertion order if an element is added a second time:

>  Note that insertion order is <i>not</i> affected if an element is <i>re-inserted</i> into the set.  (An element <tt>e</tt> is reinserted into a set <tt>s</tt> if <tt>s.add(e)</tt> is invoked when <tt>s.contains(e)</tt> would return <tt>true</tt> immediately prior to the invocation.)

Otherwise it would be better to use such a collection instead of redoing it. I could do the inertion in reversed order but end it the end its not really simpler (reverse, add, reverse)